File: src/decoders/moment.ts
import { Moment, UnknownEnumValue } from "~/models";

export function decodeMoment (value: any): Moment {
  switch (value) {
    case "matin":
      return Moment.MORNING;
    case "midi":
      return Moment.LUNCH;
    case "soir":
      return Moment.EVENING;
    default:
      throw new UnknownEnumValue("Moment", value);
  }
};

File: library/src/commonMain/kotlin/ink/literate/crowous/decoders/moment.kt
package ink.literate.crowous.decoders

import ink.literate.crowous.models.Moment

fun decodeMoment (value: String): Moment {
  return when (value) {
    "matin" -> Moment.MORNING
    "midi" -> Moment.LUNCH
    "soir" -> Moment.EVENING
    else -> throw Exception("Unknown moment: $value")
  }
}
